Review

KristjanESPERANTO opened the pull request:
I took a look at the code and suggest these changes. If you have any questions, just let me know.

- Replacing `directory-tree` by internal method - with this `npm install` is no longer needed.
- Fix name, version, contributors and description in `package.json`
- Switch LICENSE file to markdown for better readability
- Format README and add Contributing section
- Add `.gitignore` to ignore `node_modules` directory
- Replace console by Log
- Add ESLint and handle linting issues

dathbe commented:
Wow, thank you.  I just saw this and had already implemented some of it (axios replacement, e.g.).  Let me review and merge.

dathbe commented:
I'm testing your fork.  The module generally works, but `Log` doesn't seem to be logging anywhere that I can find, and local icons are not displaying (so the internal `directory-tree` method seems to not be working.  Maybe something on my system?

dathbe commented:
I've fixed the logo issue (there was an issue with the fsTree logic in node_helper.js).  But I still am not sure what's happening with `Log.log`.  I've reverted to console.log for the time being, with the old Log.log commented out.  I'm building all the new changes into a `KristjanESPERANTO` branch that I will roll into master when tested.

KristjanESPERANTO commented:
> had already implemented some of it (axios replacement, e.g.)

Okay, nice :+1:

I just rebased my fork on your master. Merge conflicts can be a mess ...

> `Log` doesn't seem to be logging anywhere that I can find

There is not much logging, but I can see the logging in the node and the browser logging.

```
[2025-03-31 23:20:31.116] [LOG]   Connecting socket for: MMM-MyStandings 
[2025-03-31 23:20:31.117] [LOG]   MMM-MyStandings helper started ... 
```

```
MyStandings notification received for NBA
MMM-MyStandings.js:266 MyStandings notification received for NFL
MMM-MyStandings.js:266 MyStandings notification received for NBA
MMM-MyStandings.js:266 MyStandings notification received for MLB_WILDCARD
MMM-MyStandings.js:266 MyStandings notification received for NBA
MMM-MyStandings.js:266 MyStandings notification received for MLB
MMM-MyStandings.js:266 MyStandings notification received for MLB_PLAYOFFS
MMM-MyStandings.js:266 MyStandings notification received for MLB
MMM-MyStandings.js:266 MyStandings notification received for NHL
....
```

> and local icons are not displaying

Ah, it could be that this doesn't work. I saw logos and thought it worked, but I didn't know they are loaded online.  I haven't looked in detail at how the local logos are loaded. We should check this.

KristjanESPERANTO commented:
> I've fixed the logo issue (there was an issue with the fsTree logic in node_helper.js).

Very good!

> But I still am not sure what's happening with Log.log. I've reverted to console.log for the time being, with the old Log.log commented out.

It's more of a cosmetic thing at the moment anyway. But it's funny that it doesn't work for you. Are you getting an error message?

> I'm building all the new changes into a KristjanESPERANTO branch that I will roll into master when tested.

Very good. There are some cosmetic changes. Perhaps it would have been better to make a smaller PR.

dathbe commented:
The changes I had made since the version you forked were not significant, so I just started from your fork and incorporated my changes.  Things seem to be working now execpt for logging.  I agree it's entirely cosmetic.  I'll play with it at some point, but for now I'm going to stick with console.log because it's working for me. I'm testing the merged branch now.  So far so good.  If i don't catch any issues, I'll merge into master.

dathbe commented:
Most changes merged into v2.5.2 with #47 .  

Will continue working on logging.

I am checking the logs in terminal using `pm2 log mm` and in my browser console.  No luck so far with `Log.log`.

KristjanESPERANTO commented:
Very good. I've had a look and have nothing to add at the moment. We can therefore close the PR here. Or would you like to leave it open as a reminder for the logging problem? :-)

dathbe commented:
I created an issue and will close the PR.  Thanks again for your help.  Based on this example, I was able to implement eslint over at MMM-MyScoreboard.

dathbe commented:
I figured out my problem.  I had my log level set to "WARN" and "ERROR".  It works great when I add "INFO", "LOG", and "DEBUG"
